formula-coroutines/src/main/java/com/instacart/formula/coroutines/FlowRuntime.kt

```diff
@@ -46,17 +49,21 @@ object FlowRuntime {
             var runtime = runtimeFactory()
 
             input.onEach { input ->
-                threadChecker.check("Input arrived on a wrong thread.")
-                if (!runtime.isKeyValid(input)) {
-                    runtime.terminate()
-                    runtime = runtimeFactory()
+//                threadChecker.check("Input arrived on a wrong thread.")
+                queue.postUpdate {
+                    if (!runtime.isKeyValid(input)) {
+                        runtime.terminate()
+                        runtime = runtimeFactory()
+                    }
+                    runtime.onInput(input)
                 }
-                runtime.onInput(input)
             }.launchIn(this)
 
             awaitClose {
-                threadChecker.check("Need to unsubscribe on the main thread.")
-                runtime.terminate()
+//                threadChecker.check("Need to unsubscribe on the main thread.")
+                queue.postUpdate {
```

Comment:
TODO: need to check for race conditions given termination is deferred and some other events could happen after it.